packages/grafana-ui/src/components/Icon/Icon.tsx

```diff
@@ -1,33 +1,93 @@
-import React from 'react';
-import { cx, css } from 'emotion';
-import { stylesFactory } from '../../themes';
+import React, { PureComponent, ComponentType } from 'react';
+import { css, cx } from 'emotion';
+
+import { stylesFactory, withTheme } from '../../themes';
+import { Themeable } from '../../types';
 import { IconType } from './types';
 
-export interface IconProps {
+type Size = 'xs' | 'sm' | 'base' | 'md' | 'lg';
+
+interface IconProps extends Themeable {
   name: IconType;
-  className?: string;
+  size?: Size;
+  color?: string;
+  type?: 'icon' | 'monochrome';
+  title?: string;
   onClick?: () => void;
   onMouseDown?: React.MouseEventHandler;
 }
+export type SvgProps = {
+  size: number;
+  color: string;
+  backgroundColor?: string;
+};
 
 const getIconStyles = stylesFactory(() => {
   return {
     icon: css`
       display: inline-block;
-      width: 16px;
-      height: 16px;
-      text-align: center;
-      font-size: 14px;
-      &:before {
+      * {
         vertical-align: middle;
       }
     `,
+    currentFontColor: css`
+      svg {
+        fill: currentColor;
+      }
+    `,
   };
 });
 
-export const Icon: React.FC<IconProps> = ({ name, className, onClick, onMouseDown }) => {
-  const styles = getIconStyles();
-  return <i className={cx(styles.icon, 'fa', `fa-${name}`, className)} onClick={onClick} onMouseDown={onMouseDown} />;
-};
+export interface IconState {
+  icon: null | ComponentType<SvgProps>;
+}
+
+class UnThemedIcon extends PureComponent<IconProps, IconState> {
```

Comment:
Minor: can we refactor this to functional component to be able to use `useTheme` hook? `withTheme` introduces some unnecessary complexity to this simple component